### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

File: src/main/java/mate/jdbc/Main.java [hunk 1]
```diff
@@ -1,7 +1,64 @@
...
+import mate.jdbc.lib.Injector;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+
 public class Main {
+    private static final Injector injector = Injector.getInstance("mate.jdbc");
+
     public static void main(String[] args) {
-        // test your code here
+        final ManufacturerDao manufacturerDao = (ManufacturerDao) injector
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:14]: You need to use service against dao in Main. 
This point was in the common mistakes, please check all your Main and change all Dao to Services
And also add ManufacturerDao to your github project when you next time will push it.

File: src/main/java/mate/jdbc/Main.java [hunk 2]
```diff
@@ -1,7 +1,64 @@
...
+                .getInstance(ManufacturerDao.class);
+        Manufacturer manufacturer = new Manufacturer();
+
+        System.out.println("Create: ");
+        manufacturer.setName("Mercedes-Benz");
+        manufacturer.setCountry("Germany");
+        Manufacturer mercedes = manufacturerDao.create(manufacturer);
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:21]: the same here

File: src/main/java/mate/jdbc/Main.java [hunk 3]
```diff
@@ -1,7 +1,64 @@
...
+        System.out.println(mercedes);
+
+        System.out.println("Get manufacturer: ");
+        Optional<Manufacturer> optionalManufacturer = manufacturerDao.get(mercedes.getId());
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:25]: the same here
Also, the Service gives you the type Manufacturer, not an Optional.

File: src/main/java/mate/jdbc/Main.java [hunk 4]
```diff
@@ -1,7 +1,64 @@
...
+        System.out.println(optionalManufacturer);
+
+        System.out.println("Get all manufacturers: ");
+        System.out.println(manufacturerDao.getAll());
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:29]: the same here
Also, you need to use Stream API forEach() to display all Manufacturers.

File: src/main/java/mate/jdbc/Main.java [hunk 5]
```diff
@@ -1,7 +1,64 @@
...
+
+        System.out.println("Update manufacturer: ");
+        manufacturer.setName("Pontiac");
+        System.out.println(manufacturerDao.update(manufacturer));
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:33]: the same here

File: src/main/java/mate/jdbc/Main.java [hunk 6]
```diff
@@ -1,7 +1,64 @@
...
+
+        System.out.println("Delete manufacturer: ");
+        boolean deleteManufacturer = manufacturerDao.delete(manufacturer.getId());
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:36]: the same here

File: src/main/java/mate/jdbc/Main.java [hunk 7]
```diff
@@ -1,7 +1,64 @@
...
+        System.out.println(deleteManufacturer);
+
+        final DriverDao driverDao = (DriverDao) injector
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:39]: the same here

File: src/main/java/mate/jdbc/Main.java [hunk 8]
```diff
@@ -1,7 +1,64 @@
...
+                .getInstance(DriverDao.class);
+        Driver driver = new Driver();
+
+        System.out.println("Create: ");
+        driver.setName("Tom");
+        driver.setLicenseNumber("111111");
+        Driver tom = driverDao.create(driver);
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:46]: the same here

File: src/main/java/mate/jdbc/Main.java [hunk 9]
```diff
@@ -1,7 +1,64 @@
...
+        System.out.println(tom);
+
+        System.out.println("Get driver: ");
+        Optional<Driver> optionalDriver = driverDao.get(tom.getId());
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:50]: the same here
Also, the Service gives you the type Driver, not an Optional.

File: src/main/java/mate/jdbc/Main.java [hunk 10]
```diff
@@ -1,7 +1,64 @@
...
+        System.out.println(optionalDriver);
+
+        System.out.println("Get all drivers: ");
+        System.out.println(driverDao.getAll());
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:54]: the same here
Also, you need to use Stream API forEach() to display all Drivers.

File: src/main/java/mate/jdbc/Main.java [hunk 11]
```diff
@@ -1,7 +1,64 @@
...
+
+        System.out.println("Update drivers: ");
+        manufacturer.setName("Alice");
+        System.out.println(driverDao.update(driver));
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:58]: the same here

File: src/main/java/mate/jdbc/Main.java [hunk 12]
```diff
@@ -1,7 +1,64 @@
...
+
+        System.out.println("Delete driver: ");
+        boolean deleteDriver = driverDao.delete(driver.getId());
```
Comment (LubusUzh) [inline src/main/java/mate/jdbc/Main.java:61]: the same here

**Review by LubusUzh — CHANGES_REQUESTED** [review]
Good job at all, just a little bit fix it!

**Review by LubusUzh — APPROVED** [review]

File: src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.service.ManufacturerService;
+
+@Dao
+public class ManufacturerServiceImpl implements ManufacturerService {
+
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/service/impl/ManufacturerServiceImpl.java:12]: remove empty line

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**